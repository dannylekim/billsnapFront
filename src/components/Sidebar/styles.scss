@import '../../variables';

.billSnap-SideBar {
  transition: 0.3s;
  position: fixed;
  height: 100vh;
  background-color: $secondary-color;
  width: 302px;
  margin-left: 0;
  left: 0;
  top: 0;
  border-radius: 0 15px 15px 0;

}
.sideLogo.navbar-brand {
  display: block;
  margin-bottom: 50px;
}
.billSnap-SideBar.sideLogo {
  color: $primary-text-color;
}
//note:
//make border color transparent
//make css background to match round
.nav-link {
  padding: 15px 8px 15px 16px;
  text-decoration: none;
  font-size: 30px;
  color: $primary-text-color !important;
  &:hover {
    background-color: rgb(106, 252, 192);
    border-color: transparent;
  }
  &.active {
    color: $secondary-color !important;
  }
}