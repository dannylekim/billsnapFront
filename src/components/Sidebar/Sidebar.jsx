import React from 'react';
import {FaUserCircle} from 'react-icons/fa';
import {Nav, NavItem, NavLink} from 'shards-react';

import './styles.scss';

export default () => {
    return (
        <Nav className={"billSnap-SideBar"} vertical={true}>
            <NavItem>
                <NavLink href="#">
                    <FaUserCircle/>
                </NavLink>
            </NavItem>
            <NavItem>
                <NavLink href="#">
                    Profile
                </NavLink>
            </NavItem>
            <NavItem>
                <NavLink href="#">
                    Settings
                </NavLink>
            </NavItem>
            <NavItem>
                <NavLink href="#">
                    Log Out
                </NavLink>
            </NavItem>
        </Nav>
    )
};