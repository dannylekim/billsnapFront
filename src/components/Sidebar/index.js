import Sidebar from "./Sidebar.jsx";
import {connect} from "react-redux";
import {setUser} from "../../redux/actions/userActions";
import { withRouter} from 'react-router-dom';

const mapDispatchToProps = (dispatch) => ({
  setUser: (user = {}) => dispatch(setUser(user)),
});

export default withRouter(connect(null, mapDispatchToProps)(Sidebar));
