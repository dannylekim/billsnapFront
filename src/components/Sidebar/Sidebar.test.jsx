import React from "react";
import Sidebar, {DEFAULT_ACTIVE_STATE} from "./Sidebar.jsx";
import SidebarComponent from "../Sidebar";
import {NavLink} from "shards-react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureStore();

const store = mockStore({});

describe("Sidebar", () => {
  let wrapper;
  let handleMockFunction, mockSetUser,mockSetActiveState, mockSetComponent;

  const STARTING_ACTIVE_STATE = {
    ...DEFAULT_ACTIVE_STATE,
    bills: true
  };

  mockSetActiveState = (activeState) => wrapper.setProps({activeState});

  beforeEach(() => {
    handleMockFunction = jest.fn();
    mockSetUser = jest.fn();
    mockSetComponent= jest.fn();
    wrapper = shallow(<Sidebar setUser={mockSetUser} activeState={STARTING_ACTIVE_STATE} setActiveState={mockSetActiveState} setComponent={mockSetComponent}/>);
  });

  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("Sidebar should match snap shot", () => {
        matches(<Sidebar activeState={STARTING_ACTIVE_STATE} setActiveState={mockSetActiveState} setComponent={mockSetComponent}/>);
      });

      it("SidebarComponent should match snap shot", () => {
        matches(
          <Provider store={store}>
            <SidebarComponent setUser={mockSetUser} activeState={STARTING_ACTIVE_STATE} setActiveState={mockSetActiveState} setComponent={mockSetComponent}  />
          </Provider>
        );
      });

      it("Sidebar should match snap shot", () => {
        const EXPECTED_STATE = {
          ...DEFAULT_ACTIVE_STATE,
          contacts: true,
        };
        wrapper = shallow(<Sidebar setUser={mockSetUser} activeState={EXPECTED_STATE} setActiveState={mockSetActiveState} setComponent={mockSetComponent} />);
        matches(wrapper);
      });
    });
    describe("interactions", () => {
      it("Navlink should call all click handles on click", () => {
        const mockHandleClick = jest.fn();
        const mockHandleLogoutClick = jest.fn();

        wrapper.instance().handleClick = mockHandleClick;
        wrapper.instance().handleLogoutClick = mockHandleLogoutClick;
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__bills")
          .simulate("click");
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__profile")
          .simulate("click");
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__contacts")
          .simulate("click");
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__settings")
          .simulate("click");
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__help")
          .simulate("click");
        wrapper
          .find(NavLink)
          .filter("#billSnap-SideBar__logout")
          .simulate("click");
        expect(mockHandleClick).toHaveBeenCalledTimes(5);
        expect(mockHandleLogoutClick).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe("function", () => {
    describe("handleClick", () => {
      it("should not change props if parameter does not exist in this props", () => {
        wrapper.instance().handleClick("starlord");
        expect(wrapper.find(NavLink).at(0).prop("active")).toBeTruthy();
      });
      it("should change props if parameter does exist in this props", () => {
        wrapper.instance().handleClick("contacts");
        expect(wrapper.find(NavLink).at(0).prop("active")).toBeFalsy();
        expect(wrapper.find(NavLink).at(2).prop("active")).toBeTruthy();
      });
    });
    describe("handleLogoutClick", () => {
      it("should clear local storage", () => {
        const EXPECTED_LENGTH = 0;
        wrapper.setProps({
          setUser: mockSetUser,
        });
        wrapper.instance().handleLogoutClick();
        const ACTUAL_LENGTH = localStorage.length;
        expect(ACTUAL_LENGTH).toEqual(EXPECTED_LENGTH);
      });
    });
  });
});
