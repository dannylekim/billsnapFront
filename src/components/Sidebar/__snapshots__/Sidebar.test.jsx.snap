// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar render snapshots ðŸ“¸ Sidebar should match snap shot 1`] = `
<nav
  className="billSnap-SideBar navbar"
  style={Object {}}
>
  <ul
    className="nav flex-column nav-tabs"
  >
    <a
      className="sideLogo navbar-brand"
    >
      Billsnap Logo
    </a>
    <div
      className="sidebar__nav-items"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          id="billSnap-SideBar__bills"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 17H6v-2h12v2zm0-4H6v-2h12v2zm0-4H6V7h12v2zM3 22l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20z"
            />
          </svg>
           Bills
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__profile"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
            />
          </svg>
           Profile
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__contacts"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           Contacts
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__settings"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
            />
          </svg>
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__help"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
           Help
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          id="billSnap-SideBar__logout"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
           Log out
        </a>
      </li>
    </div>
  </ul>
</nav>
`;

exports[`Sidebar render snapshots ðŸ“¸ Sidebar should match snap shot if hide props is true 1`] = `
<nav
  className="billSnap-SideBar navbar"
  style={
    Object {
      "display": "none",
    }
  }
>
  <ul
    className="nav flex-column nav-tabs"
  >
    <a
      className="sideLogo navbar-brand"
    >
      Billsnap Logo
    </a>
    <div
      className="sidebar__nav-items"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          id="billSnap-SideBar__bills"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 17H6v-2h12v2zm0-4H6v-2h12v2zm0-4H6V7h12v2zM3 22l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20z"
            />
          </svg>
           Bills
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__profile"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
            />
          </svg>
           Profile
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__contacts"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           Contacts
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__settings"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
            />
          </svg>
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__help"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
           Help
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          id="billSnap-SideBar__logout"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
           Log out
        </a>
      </li>
    </div>
  </ul>
</nav>
`;

exports[`Sidebar render snapshots ðŸ“¸ Sidebar should match snap shot is another row is active 1`] = `
<nav
  className="billSnap-SideBar navbar"
  style={Object {}}
>
  <ul
    className="nav flex-column nav-tabs"
  >
    <a
      className="sideLogo navbar-brand"
    >
      Billsnap Logo
    </a>
    <div
      className="sidebar__nav-items"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__bills"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 17H6v-2h12v2zm0-4H6v-2h12v2zm0-4H6V7h12v2zM3 22l1.5-1.5L6 22l1.5-1.5L9 22l1.5-1.5L12 22l1.5-1.5L15 22l1.5-1.5L18 22l1.5-1.5L21 22V2l-1.5 1.5L18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2 4.5 3.5 3 2v20z"
            />
          </svg>
           Bills
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__profile"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 11.75c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zm6 0c-.69 0-1.25.56-1.25 1.25s.56 1.25 1.25 1.25 1.25-.56 1.25-1.25-.56-1.25-1.25-1.25zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8 0-.29.02-.58.05-.86 2.36-1.05 4.23-2.98 5.21-5.37C11.07 8.33 14.05 10 17.42 10c.78 0 1.53-.09 2.25-.26.21.71.33 1.47.33 2.26 0 4.41-3.59 8-8 8z"
            />
          </svg>
           Profile
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          id="billSnap-SideBar__contacts"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           Contacts
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__settings"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
            />
          </svg>
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          id="billSnap-SideBar__help"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
            />
          </svg>
           Help
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          id="billSnap-SideBar__logout"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4"
            />
            <polyline
              points="16 17 21 12 16 7"
            />
            <line
              x1="21"
              x2="9"
              y1="12"
              y2="12"
            />
          </svg>
           Log out
        </a>
      </li>
    </div>
  </ul>
</nav>
`;
